### ISSUE TITLE
[RFE] databooks run

### ISSUE THREAD

**Issue by frenzymadness** [datarootsio/databooks#55]
I like the way databooks is able to show the content of the notebook. It might make sense to implement a `run` command which can run all cells in a notebook and show the progress in the CLI interface cell by cell. We don't need to implement the execution because it can be done by nbconvert or nbclient packages.

What do you think?

**Comment by murilo-cunha** [comment]
Glad you like it!!

I had thought about that, but in the end I dropped it because indeed there are other tools that can do exactly that (`nbconvert` and `nbclient`), and the only "gain" I see is that we could see the outputs for each cell "interactively" - we could run this separately and run `databooks show` to see the outputs no?

I'm also not a super big fan of adding a new dependency just for one feature - do you see more follow up features?

If I were to go forward I'd probably try to implement this with the stdlib (`exec`, `eval` and whatnot), but it may get hacky as well (not sure how I'd do it)

All in all, it feels like it's a bit far from databooks' JSON-centric fuctionalities.. however, I was also toying with the idea of a `jupyterminal` app - basically a [textual](https://github.com/Textualize/textual) app that runs Jupyter on your terminal, which I think that this `run` functionality would fit really well there.

What do you think?

**Comment by frenzymadness** [comment]
If it means that the dependency tree would grow significantly, we can make it an optional feature. Like, if you want to use `databooks run` install `databooks[run]` which would install nbclient/nbconvert or something else. I don't think we are able to implement the functionality just by using eval and or exec.

I took a look at dependencies around nbclient/nbconvert and it seems that Jupyter notebook already depends on nbconvert so people working with notebooks would not need to install anything new but the truth is that this will probably change with notebook 7 which is now in alpha.

**Comment by murilo-cunha** [comment]
> we can make it an optional feature

Yeah that's true. After some thought I'm inclined to give it a go! 🚀 

> would not need to install anything new

Do you think that's a big issue? If we make it optional I don't think it's a big big deal. I'm more thinking of conflicting dependencies (for people using jupyter and databooks).

It feels like `nbclient` would be the way to go? From the docs it feels like a subset of `nbconvert`? `nbclient` doesn't seem to release breaking changes often, do you think we would generate dependency conflicts?
